src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -247,8 +249,11 @@ def update_argument(self, param_name, argtype):
         arg.type.update(other=argtype)
 
     def _resolve_default_value_from_cfg_file(self, arg, overrides):
+        if arg.type.required == None:
+            arg.type.required = arg.type.settings.get('required', False)
```

Comment:
This will be used as part of an editor experience where we are offering completion proposals based on the CLI's command table and its completers. The querying is done on-the-fly as part of the editor interaction, not before. (Just invited you to our repository, the still early code using this is in service/azservice/__main__.py.)